src/utils/media-devices-manager.js

```diff
@@ -191,6 +200,36 @@ export default class MediaDevicesManager {
     }
   }
 
+  /**
+   * Webkit issue on iOS>13: https://bugs.webkit.org/show_bug.cgi?id=213853
+   */
+  _iOSTrackMutedHack() {
+    const iOSVersion = parseInt(browser.version.split(".")[0]);
+    if (
+      ["iOS", "Mac OS"].includes(detectedOS) &&
+      ["safari", "ios"].includes(browser.name) &&
+      iOSVersion >= IOS_MUTED_HACK_MIN_VERSION
+    ) {
+      this.audioTrack.onmute = this._muteStateChanged;
+      this.audioTrack.onunmute = this._muteStateChanged;
+    }
+  }
+
+  _muteStateChanged() {
+    if (this._audioTrack && this._audioTrack.muted) {
+      if (!this.reconnectIfUnmuted) {
+        NAF.connection.adapter.enableMicrophone(false);
```

Comment:
You need to update the react ui as well to indicate the mic is muted and even disabled until it recreate the audio track.

Reply:
That call will update the UI too by a store property update.

Reply:
Ah this is fine then, I'm not familiar with the new code. The enableMicrophone from the janus adapter doesn't do such thing, so I wasn't expecting it to trigger an UI update here.